packages/peregrine/lib/talons/MagentoRoute/useMagentoRoute.js

```diff
@@ -40,7 +40,7 @@ export const useMagentoRoute = () => {
                         const nextMap = new Map(prevMap);
                         const nextValue = routeError
                             ? { hasError: true, routeError }
-                            : { component, id };
```

Comment:
Instead of passing to this map, consider using a `reactiveVar`: https://www.apollographql.com/docs/react/local-state/reactive-variables/ that provides access through apollo client.